bazel_tools/scala-binary-rootpath.patch

```diff
@@ -0,0 +1,29 @@
+diff --git a/scala/private/phases/phase_default_info.bzl b/scala/private/phases/phase_default_info.bzl
+index 53f0568..8424cc8 100644
+--- a/scala/private/phases/phase_default_info.bzl
++++ b/scala/private/phases/phase_default_info.bzl
+@@ -7,6 +7,7 @@
+ def phase_default_info(ctx, p):
+     executable = None
+     files = []
++    direct = None
+     runfiles = []
+ 
+     phase_names = dir(p)
+@@ -18,6 +19,7 @@ def phase_default_info(ctx, p):
+         if hasattr(phase, "executable"):
+             if executable == None:
+                 executable = phase.executable
++                direct = [executable]
+             else:
+                 fail("only one executable may be provided")
+ 
+@@ -31,7 +33,7 @@ def phase_default_info(ctx, p):
+         external_providers = {
+             "DefaultInfo": DefaultInfo(
+                 executable = executable,
+-                files = depset(transitive = files),
++                files = depset(direct = direct, transitive = files),
```

Comment:
Relatedly, might be worth checking if `runfiles` contain `executable` as well. See https://groups.google.com/g/bazel-discuss/c/oEyEIr7vaF4/m/0E5QaK46AQAJ .

Reply:
Thanks for the tip! I’ve added it in the runfiles depset as well.